### PR TITLE
Map sessions list view

### PR DESCRIPTION
**Pull request by jankrzempek**
TASK: https://trello.com/c/WCb7N7zX/1841-sessions-list-on-the-map

- Implemented a list view for both mobile and desktop layouts.
- Scroll functionality has been designed according to the specifications in Figma. Please note that the scroll arrows are not included in this PR.
- Average real value to be added later, along with color for threshold dot

**Desktop Functionality:**

- Clicking on a tile will open the corresponding session in a modal view.
- Hovering over a tile will highlight the session. Careful observation is recommended, as clustered sessions will also be highlighted.

**Mobile Functionality:**

- Tapping a session tile will redirect the user to the calendar.
- The interface will display a list of sessions, the total number of sessions, and an 'X' button to close the view. No additional features are included in this PR.

### CODE REVIEW

File: app/javascript/react/store/fixedSessionsSelectors.ts [hunk 1]
```diff
@@ -21,4 +34,29 @@ const selectFixedSessionsPoints = createSelector(
...
+  (fixedSessionsState): SessionList[] =>
+    fixedSessionsState.sessions.map(
+      ({ id, title, startTimeLocal, endTimeLocal, streams }) => {
+        const firstStream = streams[Object.keys(streams)[0]];
+
+        return {
+          id,
+          title,
+          sensorName: firstStream.sensorName,
+          // Please change averageValue once backend will be ready
```
Comment (JoannaThelema) [inline app/javascript/react/store/fixedSessionsSelectors.ts:48]: Let's add this info to some ticket or create new one 💪

File: app/javascript/react/store/mobileSessionsSelectors.ts [hunk 2]
```diff
@@ -56,8 +57,28 @@ const selectMobileSessionsPoints = createSelector(
...
+  [selectMobileSessionsState],
+  (mobileSessionsState): SessionList[] =>
+    mobileSessionsState.sessions.map(({ id, title, startTimeLocal, endTimeLocal, streams }) => {
+      const firstStream = streams[Object.keys(streams)[0]];
+
+      return {
+        id,
+        title,
+        sensorName: firstStream.sensorName,
+        // Please change averageValue once backend will be ready
```
Comment (JoannaThelema) [inline app/javascript/react/store/mobileSessionsSelectors.ts:70]: same

File: app/javascript/react/components/Map/Markers/SessionType.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { FixedStream } from "../../../types/fixedStream";
+import { LatLngLiteral } from "../../../types/googleMaps";
+
+type Point = LatLngLiteral & { streamId: string };
+
+export type Session = {
+  id: number;
+  lastMeasurementValue: number;
+  point: Point;
+};
```
Comment (JoannaThelema) [inline app/javascript/react/components/Map/Markers/SessionType.ts:10]: We already have this file in: `app/javascript/react/types/sessionType.ts`

**Comment by JoannaThelema** [comment]
In the initial map view:

Screenshot 2024-06-20 at 08 43 48

**Comment by JoannaThelema** [comment]
On tiles, you always have 00:00 time. 

Screenshot 2024-06-20 at 08 47 49

**Comment by JoannaThelema** [comment]
I think the search button should disappear. 
Screenshot 2024-06-20 at 08 51 39

**Review by JoannaThelema — COMMENTED** [review]
I found a few bugs but otherwise looks good 🙂

File: app/javascript/react/components/SectionButton/sectionButton.style.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  background-color: white;
+  border-radius: 10px;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  margin-bottom: 0.4em;
+  box-sizing: border-box;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 85px;
```
Comment (evulins) [inline app/javascript/react/components/SectionButton/sectionButton.style.tsx:14]: I would use rem also here for width and height : 8.5rem and 4.3rem

File: app/javascript/react/components/SectionButton/sectionButton.style.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  right: 1.25em;
+  border: none;
+`;
+
+const Title = styled(H6)``;
+
+const Image = styled.img`
+  width: 10px;
+  height: 12px;
+  margin-bottom: 0.25em;
```
Comment (evulins) [inline app/javascript/react/components/SectionButton/sectionButton.style.tsx:27]: same here :)

File: app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+const ImageButton = styled.button`
+  background: none;
+  border: none;
+  cursor: pointer;
+  padding: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-right: 0.625em;
```
Comment (evulins) [inline app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx:25]: Could you please use rem?

File: app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  margin-top: 1.25em;
+  overflow-y: auto;
+  max-height: 100%;
+`;
+
+const SessionListStyled = styled.div`
+  width: 100%;
+  overflow-y: auto;
+  margin-left: 1.25em;
+  margin-right: 1.25em;
```
Comment (evulins) [inline app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx:61]: same here in every styled component. 

 Is there any specific reason for using the em unit in this PR?

**Review by evulins — COMMENTED** [review]
Great work 👍 🥇 

Here are  my comments:
- Could you please change `em` unit to `rem`? 
- Is there a reason why you named files with lower case letter e.g, sessionListTile? In ReactJS for components in this project we use capital letter
-The font style for the session name is slightly different than the design.

File: app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.style.tsx [hunk 8]
```diff
@@ -1,9 +1,25 @@
-import styled from "styled-components";
+import styled, { keyframes, css } from "styled-components";
 
 interface MarkerProps {
   color: string;
+  shouldPulse?: boolean;
```
Comment (evulins) [inline app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.style.tsx:5]: use here $ before:` $shouldPulse?: boolean;`

File: app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.style.tsx [hunk 9]
```diff
@@ -40,6 +56,7 @@ const MarkerCircle = styled.div<MarkerProps>`
   border-radius: 50%;
   background-color: ${(props) => props.color};
   pointer-events: none;
+  animation: ${props => props.shouldPulse ? css`${pulseAnimation} 2s infinite` : 'none'};
```
Comment (evulins) [inline app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.style.tsx:59]: animation: ${props => props.$shouldPulse ? css`${pulseAnimation} 2s infinite` : 'none'};

File: app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.tsx [hunk 10]
```diff
@@ -4,14 +4,15 @@ import { DataContainer, MarkerCircle, MarkerContainer } from "./SessionDotMarker
...
   onClick: () => void;
 }
 
-const SessionDotMarker = ({ color, onClick }: MarkerProps) => {
+const SessionDotMarker = ({ color, shouldPulse = false, onClick }: MarkerProps) => {
   return (
     <MarkerContainer onClick={onClick}>
       <DataContainer>
-        <MarkerCircle color={color} />
+        <MarkerCircle color={color} shouldPulse={shouldPulse}/>
```
Comment (evulins) [inline app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.tsx:15]: <MarkerCircle color={color} $shouldPulse={shouldPulse}/>

Comment (evulins) [reply app/javascript/react/components/Map/Markers/SessionDotMarker/SessionDotMarker.tsx:15]: this should help with error in console

File: app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx (hunk 7 above)
Comment (jankrzempek) [reply app/javascript/react/components/SessionsListView/MobileSessionList/mobileSessionList.style.tsx:61]: no, my mistake, i was thinking about rem

**Review by evulins — APPROVED** [review]
🚀 🥇

**Review by JoannaThelema — APPROVED** [review]

**Pull request merged**